infrastructure/k8s/readme.md

```diff
@@ -0,0 +1,290 @@
+ with advanced network setup to enabled communication between ACI (Azure container instance) and the cluster.
+
+Enable ACI for your subscription.
+```sh
+az provider register --namespace Microsoft.ContainerInstance
+```
+
+Create resources group 
+```sh
+az group create --name ${res} --location ${region}
+```
+
+Create a VNET with the two subnets that are going to be used.
+```sh
+az network vnet create \
+    --resource-group ${res} \
+    --name coverServiceVnet \
+    --address-prefixes 10.0.0.0/8 \
+    --subnet-name coverServiceSubnet \
+    --subnet-prefix 10.240.0.0/16
+```
+
+And an additional subnet to connect to ACI.
+```sh
+az network vnet subnet create \
+    --resource-group ${res} \
+    --vnet-name coverServiceVnet \
+    --name coverServiceVirtualNodeSubnet \
+    --address-prefix 10.241.0.0/16
+```
+
+To connect to ACI you need a service principal assignment. So create the principal first.
+```sh
+az ad sp create-for-rbac --skip-assignment
+```
+
+From the output of the previous command we need to save the application id and password (app secret).
+```sh
+APPID=<id>
+PASSWORD=<passwd>
+```
+
+Get the Azure id for the virtual network to allow it access to ACI.
+```sh
+VNETID=$(az network vnet show --resource-group ${res} --name coverServiceVnet --query id -o tsv)
+```
+
+Create the access rule.
+```sh
+az role assignment create --assignee $APPID --scope $VNETID --role Contributor
+```
+
+Get the Azure subnet id.
+```sh
+SUBNET=$(az network vnet subnet show --resource-group ${res} --vnet-name coverServiceVnet --name coverServiceSubnet --query id -o tsv)
+```
+
+Create the cluster and wait for it to create the 3 nodes in the standard cluster (it takes a bit of time, so go ahead 
+and get yourself a cup of coffee).
+```sh
+az aks create \
+    --resource-group ${res} \
+    --name ${ksname} \
+    --node-count 3 \
+    --node-vm-size Standard_DS3_v2 \
+    --kubernetes-version ${version} \
+    --network-plugin azure \
+    --service-cidr 10.0.0.0/16 \
+    --dns-service-ip 10.0.0.10 \
+    --docker-bridge-address 172.17.0.1/16 \
+    --vnet-subnet-id $SUBNET \
+    --service-principal $APPID \
+    --client-secret $PASSWORD 
+```
+
+Enable the virtual node in your cluster.
+````sh
+az aks enable-addons \
+    --resource-group ${res} \
+    --name ${ksname} \
+    --addons virtual-node \
+    --subnet-name coverServiceVirtualNodeSubnet
+````
+
+Configure kubeclt to connect to the new cluster
+```sh
+az aks get-credentials --resource-group ${res} --name ${ksname}
+```
+
+Verify that you are connected to the cluster now.
+```sh
+kubectl get nodes
+```
+
+## Helm
+We are going to use https://helm.sh/ to install ingress and cert-manager into the cluster setup. Note that we here are using helm version 3. We also install the kubectx helper tool as it makes switching cluster and namespaces easier.
+```sh
+brew install helm
```

Comment:
Only mac ?

Reply:
Yes, as i don't know how to install is in Windows or Linux.